test/framework/toy_build.py

```diff
@@ -2707,6 +2708,25 @@ def grab_gcc_rpath_wrapper_args():
         write_file(toy_ec, toy_ec_txt)
         self.test_toy_build(ec_file=toy_ec, extra_args=['--rpath', '--experimental'], raise_error=True)
 
+    def test_toy_rpath_filter(self):
+        """Test toy build using --rpath --rpath-filter and --rpath --rpath-filter --filter-rpath-sanity-libs."""
+
+        # test sanity error when --rpath-filter is used to filter a required library
+        # In this test, libtoy.so will be linked, but not RPATH-ed due to the --rpath-filter
+        # Thus, the RPATH sanity check is expected to fail with libtoy.so not being found
+        test_ecs = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'easyconfigs', 'test_ecs')
+        toy_ec = os.path.join(test_ecs, 't', 'toy2', 'toy2-0.0.eb')
+        error_pattern = r"Sanity check failed\: Library libtoy\.so not found"
+        self.assertErrorRegex(EasyBuildError, error_pattern, self.test_toy_build, ec_file=toy_ec,
+                              extra_args=['--rpath', '--experimental', '--rpath-filter=.*libtoy.*'],
+                              name='toy2', raise_error=True, verbose=False)
```

Comment:
Also run again without `--rpath-filter`, for completeness sake?
It's generally a good idea to make sure that the test setup is done fully correctly...

```suggestion
        args = ['--rpath', '--experimental', '--rpath-filter=.*libtoy.*']
        self.assertErrorRegex(EasyBuildError, error_pattern, self.test_toy_build, ec_file=toy_ec,
                              extra_args=args, name='toy2', raise_error=True, verbose=False)
        # works fine if --rpath-filter is not used (since then libtoy is RPATH'ed)
        self.test_toy_build(ec_file=toy_ec, name='toy2', extra_args=args, raise_error=True)
```

You will need to clean up the install dir to make sure the next part of the test still works as expected.

Reply:
Why do we need to clean? A rebuild will clean the installdir anyway, doesn't it?

Reply:
Yes, indeed